datafusion/expr/src/expr.rs

```diff
@@ -3067,4 +3078,19 @@ mod test {
             rename: opt_rename,
         }
     }
+
+    #[test]
+    fn test_size_of_expr() {
+        // because Expr is such a widely used struct in DataFusion
+        // it is important to keep its size as small as possible
+        //
+        // If this test fails when you change `Expr`, please try
+        // `Box`ing the fields to make `Expr` smaller
+        // See https://github.com/apache/datafusion/issues/14256 for details
+        assert_eq!(size_of::<Expr>(), 112);
+        assert_eq!(size_of::<ScalarValue>(), 64);
+        assert_eq!(size_of::<DataType>(), 24); // 3 ptrs
```

Comment:
does this test pass in debug and release modes?

Reply:
It does:

```shell
$ cargo test --release -p datafusion-expr -- test_size_of_expr
...

running 1 test
test expr::test::test_size_of_expr ... ok
```